### PR TITLE
docs(samples): add a new sample for parallel append

### PR DESCRIPTION
**Pull request by yayi-google**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: samples/snippets/src/test/java/com/example/bigquerystorage/ParallelWriteCommittedStreamIT.java [hunk 1]
```diff
@@ -83,14 +83,15 @@ public void setUp() {
 
   @After
   public void tearDown() {
+    bigquery.delete(TableId.of(GOOGLE_CLOUD_PROJECT, datasetName, tableName));
     bigquery.delete(
         DatasetId.of(GOOGLE_CLOUD_PROJECT, datasetName), DatasetDeleteOption.deleteContents());
     System.setOut(null);
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerystorage/ParallelWriteCommittedStreamIT.java:89]: You should restore console here.

Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerystorage/ParallelWriteCommittedStreamIT.java:89]: restore stdout

Comment (yayi-google) [reply samples/snippets/src/test/java/com/example/bigquerystorage/ParallelWriteCommittedStreamIT.java]: Done.

File: samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquerystorage;
+
+// [START bigquerystorage_jsonstreamwriter_committed]
```
Comment (yirutang) [inline samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java:19]: What does this mean and is it useful?

**Review by yirutang — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java (hunk 2 above)
Comment (yayi-google) [reply samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java:19]: I don't know. I copied and pasted here...

**Comment by yayi-google** [comment]
The only one failing test is unrelated to this CL, and is removed in a separate pull request.

File: samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java (hunk 2 above)
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java:19]: this needs to be unique. Can we go with: "bigquerystorage_jsonstreamwriter_parallelcommitted"?

**Comment by stephaniewang526** [comment]
please change the subject and initial commit from `feat: ...` to `docs(samples): ...` since this is not a feat and it will go into the release notes.

File: samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java (hunk 2 above)
Comment (yayi-google) [reply samples/snippets/src/main/java/com/example/bigquerystorage/ParallelWriteCommittedStream.java]: Sounds good. Done.

**Review by stephaniewang526 — APPROVED** [review]

**Pull request merged**